style: homepage components situated

bry-gavino opened the pull request:
Finished placing components where they should be in accordance to Figma. The only difference from Figma is that there is a MapVisual.

## Screenshots

<img width="1679" alt="Screen Shot 2020-10-13 at 11 57 23 PM" src="https://user-images.githubusercontent.com/60284767/95954608-6e3f9d00-0db0-11eb-8b11-f173b4970ebf.png">
<img width="657" alt="Screen Shot 2020-10-13 at 11 57 46 PM" src="https://user-images.githubusercontent.com/60284767/95954611-6f70ca00-0db0-11eb-9c44-3100474bb313.png">


CC: @kwfk

bry-gavino committed:
style: homepage components situated

finished placing components where they should be in accordance to figma

bry-gavino committed:
docs: fixed env and local package

kwfk commented on components/Map.tsx:
```diff
@@ -3,8 +3,8 @@ import ReactMapGL from 'react-map-gl';
-    width: 100,
-    height: 100,
+    width: 50000000, // large number to fit the box?
+    height: 400,
```
The reason why we were getting type errors from `ViewportProps` previous is because we only implicitly declared what type our state variable should accept. When we only had "100%" the state variable was only expecting a string but since the `onViewportChange` updated the state variable with a number type, Typescript raised an error. In order to remedy this, we want to explicitly tell the state variable what the object type should be. We can do this by doing the follow:

create a new `type` outside our component
```typescript
type ViewportStateProps = {
    width: number | string;
    height: number | string;
}
```
Set the state variable to the type:
```typescript
const [viewport, setViewport] = useState<ViewportStateProps>({...})
```
You will get errors because the explicitly set type doesn't include all the other keys in the state variable object. We will fix it in the next steps.

Because we know that `onViewportChange` will update the state, we want to also include the type of `ViewportProps`. We can do this by creating a union with our changed type with `react-map-gl`'s type.
We first want to import the `ViewportProps` type:
```typescript
import ReactMapGL, { ViewportProps } from 'react-map-gl'
```
then union our new type object with `ViewportProps` to get this:
```typescript
const [viewport, setViewport] = useState<ViewportStateProps | ViewportProps>({...})
```

kwfk commented on pages/index.tsx:
```diff
@@ -1,24 +1,99 @@
+import Box from '@material-ui/core/Box';
+import {
+  Button,
+  CircularProgress,
```
let's remove this import because we are not using this component anymore

kwfk commented on pages/index.tsx:
```diff
@@ -1,24 +1,99 @@
+import SearchIcon from '@material-ui/icons/Search';
+import styles from '../styles/Home.module.css';
 
 const MapVisual = dynamic(() => import('../components/Map'), {
```
I think just calling this `Map` will be cleaner

kwfk commented on pages/index.tsx:
```diff
@@ -1,24 +1,99 @@
 import dynamic from 'next/dynamic';
-import Layout from 'components/Layout';
-import { Button, CircularProgress } from '@material-ui/core';
+import Box from '@material-ui/core/Box';
```
let's use `div`s instead of `Box`

kwfk replied:
You can simply find and replace all the `Box` with `div`

kwfk commented on styles/Home.module.css:
```diff
@@ -121,3 +121,85 @@
+
+.pageTitle {
+  font-weight: 800;
+  font-size: 50px;
```
for font sizes, we typically want to use em or rem. In our case, 1 rem is 16px

> While em is relative to the font-size of its direct or nearest parent, rem is only relative to the html (root) font-size.
https://css-tricks.com/confused-rem-em/

kwfk commented on pages/index.tsx:
```diff
@@ -1,24 +1,99 @@
+            Join Us
+          </Button>
+        </Box>
+      </Box>
```
Instead of having a navbar that is specific to this page, let's update the `Layout` component so that it is updated across all the pages.

kwfk commented on pages/index.tsx:
```diff
@@ -1,24 +1,99 @@
+      <Box className={styles.pageContent}>
+        <Box className={styles.searchBar}>
+          <Box>
+            <IconButton color="primary">
+              <SearchIcon />
+            </IconButton>
+            <TextField id="standard-basic" label="Explore Organizations" />
```
take a look at the `TextField` with adornment component
https://material-ui.com/components/text-fields/#icons

kwfk replied:
also let's use `variant=outlined` for now instead of a background color

kwfk replied:
also make it so that the TextField itself fills the entire width of the screen

kwfk commented on pages/index.tsx:
```diff
@@ -1,24 +1,99 @@
+          <Box className={styles.text}>SampleText3</Box>
+        </Box>
+
+        <Box className={styles.component} display="flex">
```
remove `display="flex"`

kwfk commented on pages/index.tsx:
```diff
@@ -1,24 +1,99 @@
+          </Box>
+          <Box className={styles.keywords}>
+            <Box className={styles.keyButtonSpace}>
+              <FormControl className={styles.keyword} variant="outlined">
+                <InputLabel>Keyword</InputLabel>
+                <Select>
+                  <MenuItem value={10}>Test1</MenuItem>
+                  <MenuItem value={20}>Test1</MenuItem>
+                  <MenuItem value={30}>Test1</MenuItem>
+                </Select>
+              </FormControl>{' '}
+            </Box>
+            <Box className={styles.keyButtonSpace}>
+              <FormControl className={styles.keyword} variant="outlined">
+                <InputLabel>Keyword</InputLabel>
+                <Select>
+                  <MenuItem value={10}>Test1</MenuItem>
+                  <MenuItem value={20}>Test1</MenuItem>
+                  <MenuItem value={30}>Test1</MenuItem>
+                </Select>
+              </FormControl>{' '}
+            </Box>
+            <Box className={styles.keyButtonSpace}>
+              <FormControl className={styles.keyword} variant="outlined">
+                <InputLabel>Keyword</InputLabel>
+                <Select>
+                  <MenuItem value={10}>Test1</MenuItem>
+                  <MenuItem value={20}>Test1</MenuItem>
+                  <MenuItem value={30}>Test1</MenuItem>
+                </Select>
+              </FormControl>
```
figure out how to make the selects equi-width?

bry-gavino replied:
I ended up making the buttons bigger to fit the text. lmk if you want me to find a different way to work around this!

kwfk reviewed (changes requested):
Let's make this page look like the results page instead of the landing page. Sorry should have made it clearer.
Changes will be:
- moving the map to the right 2/3 of the page
- moving the org list to the left 1/3 of the page
- putting the keywords on top of the org list instead of inside the search bar
You don't have to create the org cards for now

bry-gavino committed:
style: fixed all but map

fixed most of the PR issues but the map

bry-gavino committed:
style: resolved map kinda

bry-gavino reviewed (commented):
<img width="1675" alt="Screen Shot 2020-10-18 at 7 43 13 PM" src="https://user-images.githubusercontent.com/60284767/96395878-9cd7c200-117a-11eb-8552-cd2129025a7e.png">
<img width="574" alt="Screen Shot 2020-10-18 at 7 43 22 PM" src="https://user-images.githubusercontent.com/60284767/96395882-9ea18580-117a-11eb-9287-1271302e4d37.png">

kwfk commented on components/Layout.tsx:
```diff
@@ -1,49 +1,35 @@
-        | <a href="/api/users">Users API</a>
-      </nav>
-    </header>
+    <div className={styles.pageTitle}>
```
we want to keep the Head component because that changes the title on the tab bar

kwfk commented on components/Layout.tsx:
```diff
@@ -1,49 +1,35 @@
-    </header>
+    <div className={styles.pageTitle}>
+      {title}
+      <div>
```
let's continue using the header and nav tags here

kwfk commented on components/Layout.tsx:
```diff
@@ -1,49 +1,35 @@
+      </div>
+    </div>
     {children}
-    <footer>
```
Keep the footer as well

kwfk commented on pages/index.tsx:
```diff
@@ -1,23 +1,83 @@
+        <TextField
+          id="outlined-size-small"
+          placeholder="Explore Organizations"
+          size="normal"
```
no such size as `normal`. we can just remove this

kwfk commented on pages/index.tsx:
```diff
@@ -1,23 +1,83 @@
+              <div className={styles.keyButtonSpace}>
+                <FormControl className={styles.keyword} variant="outlined">
+                  <InputLabel>Keyword</InputLabel>
+                  <Select>
```
need to add a `label` prop that is the same as the `InputLabel` value

bry-gavino replied:
I'm not exactly sure if I did these correctly, but I just wanted to ask just in case :))!
<img width="396" alt="Screen Shot 2020-10-21 at 10 27 55 AM" src="https://user-images.githubusercontent.com/60284767/96755870-1e506f80-1388-11eb-8e9f-6061d8eb4d1d.png">
<img width="358" alt="Screen Shot 2020-10-21 at 10 28 02 AM" src="https://user-images.githubusercontent.com/60284767/96755882-20b2c980-1388-11eb-9181-a7e739f80746.png">

bry-gavino replied:
I ended up pushing a commit now just in case you wanted to look at the changes immediately instead of waiting for me to push, but lmk if I still need to fix more things!

kwfk replied:
ah it would just be `label="Keyword"`

kwfk replied:
oh you want the `label` prop in the Select component

kwfk replied:
the comment i add usually refers to the last line of the preview

bry-gavino replied:
ohhhhh okie gotcha, thanks for letting me know!!

kwfk commented on pages/index.tsx:
```diff
@@ -1,23 +1,83 @@
+                <FormControl className={styles.keyword} variant="outlined">
+                  <InputLabel>Keyword</InputLabel>
+                  <Select>
+                    <MenuItem value={10}>Test1</MenuItem>
```
add a `None` value here

kwfk commented on pages/index.tsx:
```diff
@@ -1,23 +1,83 @@
+                    <MenuItem value={20}>Test1</MenuItem>
+                    <MenuItem value={30}>Test1</MenuItem>
+                  </Select>
+                </FormControl>{' '}
```
remove space

kwfk commented on components/AdminDash.tsx:
```diff
@@ -0,0 +1,54 @@
+import Link from 'next/link';
```
let's move this to a separate branch and PR

kwfk reviewed (changes requested):
almost there!

bry-gavino committed:
style: fixed more of PR

kwfk committed:
fix(layout and home): minor changes to keyword select and layout styling

Fixed keyword select with labeling issue and added None option. Revert Navbar(Layout) to include
navigation links, header, and footer.

kwfk committed:
Merge branch 'main' into bryanna/homepage-components2

kwfk committed:
Merge branch 'main' into bryanna/homepage-components2

kwfk reviewed (approved):
LGTM
